fence/resources/admin/admin_users.py

```diff
@@ -150,19 +162,236 @@ def add_user_to_projects(current_session, username, projects=None):
+                    == google_proxy_group_f.id
+                )
+                .all()
+            )
+            for row in gpg_to_gbag:
+                current_session.delete(row)
+            capp.logger.debug("Deleting rows in user_google_account_to_proxy_group...")
+            uga_to_pg = (
+                current_session.query(UserGoogleAccountToProxyGroup)
+                .filter(
+                    UserGoogleAccountToProxyGroup.proxy_group_id
+                    == google_proxy_group_f.id
+                )
+                .all()
+            )
+            for row in uga_to_pg:
+                current_session.delete(row)
+            capp.logger.debug("Deleting rows in user_google_account...")
+            uga = (
+                current_session.query(UserGoogleAccount)
+                .filter(UserGoogleAccount.user_id == user.id)
+                .all()
+            )
+            for row in uga:
+                current_session.delete(row)
+            capp.logger.debug("Deleting row in google_proxy_group...")
+            current_session.delete(google_proxy_group_f)
+            current_session.commit()
+            capp.logger.info(
+                "Records for Google proxy group {} successfully cleared from Fence "
+                "database, along with associated user Google accounts.".format(
+                    gpg_email
+                )
+            )
+            capp.logger.info("Done with Google deletions.")
+    else:
+        raise UnavailableError(
+            "Error: Google unable to delete proxy group {}. Aborting".format(gpg_email)
+        )
+
+
 def delete_user(current_session, username):
     """
     Remove a user from both the userdatamodel
-    and the assciated storage for that project/bucket.
+    and the associated storage for that project/bucket.
     Returns a dictionary.
+
+    The Fence db may not always be in perfect sync with Google.  We err on the
+    side of safety (we prioritise making sure the user is really cleared out of
+    Google to prevent unauthorized data access issues; we prefer cirrus/Google
+    over the Fence db as the source of truth.) So, if the Fence-Google sync
+    situation changes, do edit this code accordingly.
     """
-    response = us.delete_user(current_session, username)
-    if response["result"] == "success":
-        providers = response.get("providers", [])
-        for provider in providers:
-            capp.storage_manager.delete_user(provider.backend, response["user"])
 
-        return {"result": "success"}
+    capp.logger.debug("Beginning delete user.")
+
+    with GoogleCloudManager() as gcm:
+        # Delete user's service accounts, SA keys, user proxy group from Google.
+        # Noop if Google not in use.
+
+        user = query_for_user(session=current_session, username=username)
+        if not user:
+            raise NotFound("".join(["user name ", username, " not found"]))
```

Comment:
```suggestion
            raise NotFound("user name {} not found".format(username))
```

I find this just a little easier to read but it's up to you